pkg/engine/shield/manager_test.go

```diff
@@ -0,0 +1,121 @@
+package shield
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/simimpact/srsim/tests/mock"
+)
+
+// Unit Tests for HasShield()
+func TestHasShieldWhenShield(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	events := &event.System{}
+	attr := mock.NewMockAttribute(mockCtrl)
+	manager := New(events, attr)
+
+	target := key.TargetID(1)
+	shield := &Instance{name: "Shield", hp: 0.0}
+	manager.targets[target] = append(manager.targets[target], shield)
+
+	isTrue := manager.HasShield(target, shield.name)
+	assert.Equal(t, true, isTrue)
+}
+
+func TestHasShieldWhenNoShield(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	events := &event.System{}
+	attr := mock.NewMockAttribute(mockCtrl)
+	manager := New(events, attr)
+
+	target := key.TargetID(1)
+
+	isTrue := manager.HasShield(target, key.Shield("Shield"))
+	assert.Equal(t, false, isTrue)
+}
```

Comment:
nit: should have case of has shields but not one of key time (`HasShield` false for non-empty shield array)